janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -6165,4 +6165,234 @@ private void testWritingAndReading(File f) {
+        a.property("vtName","A");
+        JanusGraphVertex b = tx.addVertex();
+        b.property("vtName","B");
+        //add Edge
+        a.addEdge("egLabel",b,"assocKind",1);
+        b.addEdge("egLabel",a,"assocKind",2);
+        newTx();
+
+        RelationType t = mgmt.getRelationType("egLabel");
+        RelationTypeIndex relationIndex = mgmt.getRelationIndex(t,indexName);
+        assertEquals(SchemaStatus.ENABLED, relationIndex.getIndexStatus());
+        
+        //asserting before reindex
+        assertEquals(0,countOfEdges("vtName", "A", "IN", "egLabel", "assocKind", 1, false));
+        assertEquals(1,countOfEdges("vtName", "A", "IN", "egLabel", "assocKind", 2, false));
+        assertEquals(1,countOfEdges("vtName", "A", "OUT", "egLabel", "assocKind", 1,false));
+        assertEquals(0,countOfEdges("vtName", "A", "OUT", "egLabel", "assocKind", 2,false));
+        assertEquals(1,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 1, false));
+        assertEquals(0,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 2, false));
+        assertEquals(0,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 1,false));
+        assertEquals(1,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 2,false));
+
+        //Reindexing
+        mgmt.updateIndex(relationIndex, SchemaAction.REINDEX).get();
+        finishSchema();
+
+        relationIndex = mgmt.getRelationIndex(t,indexName);
+        assertEquals(SchemaStatus.ENABLED, relationIndex.getIndexStatus());
+
+        //asserting after reindex
+        assertEquals(0,countOfEdges("vtName", "A", "IN", "egLabel", "assocKind", 1, false));
+        assertEquals(1,countOfEdges("vtName", "A", "IN", "egLabel", "assocKind", 2, false));
+        assertEquals(1,countOfEdges("vtName", "A", "OUT", "egLabel", "assocKind", 1,false));
+        assertEquals(0,countOfEdges("vtName", "A", "OUT", "egLabel", "assocKind", 2,false));
+        assertEquals(1,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 1, false));
+        assertEquals(0,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 2, false));
+        assertEquals(0,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 1,false));
+        assertEquals(1,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 2,false));
+    }
+	
+    private void initReindexSchema() {
+        if(!mgmt.containsEdgeLabel("egLabel")){
+            mgmt.makeEdgeLabel("egLabel").multiplicity(Multiplicity.MULTI).make();
+        }
+
+        if(!mgmt.containsPropertyKey("vtName")) {
+            mgmt.makePropertyKey("vtName").dataType(String.class).cardinality(Cardinality.SINGLE).make();
+        }
+		
+        if(!mgmt.containsPropertyKey("assocKind")) {
+            mgmt.makePropertyKey("assocKind").dataType(Integer.class).cardinality(Cardinality.SINGLE).make();	
+        }
+    }
+    
+    //To make sure no self links created
+    public void assertEdgeCounts(boolean isReindexed) {
```

Comment:
When you put an ancillary method which is for one or more specific scenarios, you'd better not use a seemingly generic name like "assertEdgeCounts".

Reply:
testcase removed as this seems to be unnecessary for this PR